src/color.rs

```diff
@@ -420,6 +420,153 @@ impl Color {
         self.alpha = alpha;
         self
     }
+
+    /// Mix this color with the other given color, with the given weighting .
+    /// 
+    /// The weight determines the amount of the first color that will be used
+    /// in mixing as a percentage. So 50% means use each color equally, while 
+    /// 25% indicates to use one-quarter of this color, and seventy-five percent 
+    /// of the second.
+    /// 
+    /// The alpha channel of both colors are also combined in this way.
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// 
+    /// let c1 = Color::rgba(18., 55., 125., 1.0);
+    /// let mixed = c1.mix(&Color::rgba(125., 33., 200., 0.8), 40);
+    /// assert_eq!(mixed, Color::rgba(72., 44., 163., 0.88));
+    /// ```
```

Comment:
(You may want to read through my comments below before this one because otherwise the syntax I used in `mix` may not make sense.)

Reading through this, I got the feeling that the weight parameter could use some more illustration. This might be a great place to put a full example program. There are a couple of places where we do this already. For example, [the complete example in the `set_pen_size` method](https://docs.rs/turtle/1.0.0-rc.1/turtle/struct.Turtle.html#method.set_pen_size) shows an example of a cool effect you can accomplish that method. 

I think a really good practical example to add here would be one where you create a turtle, set it to different mixes of a color and show the affect of the weight parameter. 

Something like the following maybe? (This should be marked as `no_run` if added)

```rust
extern crate turtle;

use turtle::{Color, Turtle};

fn main() {
    let mut turtle = Turtle::new();
    turtle.drawing_mut().set_title("Mixing colors!");
    turtle.set_pen_size(5.0);

    let red: Color = "red".into();

    // This will draw a red line
    turtle.set_pen_color(red);
    turtle.forward(100.0);
    turtle.right(90.0);

    // This will draw a purple line because we have equally mixed red with blue
    turtle.set_pen_color(red.mix("blue", 50));
    turtle.forward(100.0);
    turtle.right(90.0);

    // This will draw a line that is 25% red and 75% blue (a red-ish dark blue color)
    turtle.set_pen_color(red.mix("blue", 25));
    turtle.forward(100.0);
    turtle.right(90.0);

    // This will draw a line that is 75% red and 25% blue (medium red violet)
    turtle.set_pen_color(red.mix("blue", 75));
    turtle.forward(100.0);
    turtle.right(90.0);
}
```

This produces the following:

![turtle color mixing](https://github.com/sunjay/turtle/raw/gh-pages/assets/images/docs/color_mixing.png)

### Markdown for this image
```markdown
![turtle color mixing](https://github.com/sunjay/turtle/raw/gh-pages/assets/images/docs/color_mixing.png)
```

Feel free to come up with your own example too if you'd like to illustrate something cool. :) 

(If you decide to do that, I can upload the image for you and give you the URL at your request)

Reply:
I'll use your example as it perfectly illustrates the point without being too complex. The reader can easily attach each instruction in the example code to a part of the image